packages/server/api/src/app/ee/custom-domains/custom-domain.service.ts

```diff
@@ -33,17 +70,61 @@ export const customDomainService = {
             platformId: request.platformId,
         })
     },
-    async create(request: {
-        domain: string
+    async verifyDomain(request: {
         platformId: string
-    }): Promise<CustomDomain> {
-        const customDomain = customDomainRepo.create({
-            id: apId(),
-            domain: request.domain,
+        id: string
+    }): Promise<boolean> {
+        const customDomain = await customDomainRepo
+            .createQueryBuilder('custom_domain')
+            .where({
+                id: request.id,
+            })
+            .getRawOne()
+        const hostnameDetails = await cloudflareHostnameServices.getHostnameDetails(customDomain.custom_domain_domain)
+        const patchResult = await cloudflareHostnameServices.update(hostnameDetails.data.result.id)
+        const status = patchResult.data.result.status
+
+        await customDomainRepo.update({
             platformId: request.platformId,
+            id: request.id,
+        }, {
             status: CustomDomainStatus.ACTIVE,
         })
-        return customDomainRepo.save(customDomain)
+
+        return status
+    },
+    async create(request: {
+        domain: string
+        platformId: string
+        ssl: SSLParams
+    }): Promise<CustomDomain> {
+        try {
+            const customDomain = customDomainRepo.create({
+                id: apId(),
+                domain: request.domain, 
+                platformId: request.platformId,
+                status: CustomDomainStatus.PENDING,
+            })
+            
+            const edition = getEdition()
+            if (edition === ApEdition.CLOUD) {
+                await cloudflareHostnameServices.create({ 
+                    hostname: request.domain, 
+                    ssl: request.ssl, 
+                })
+            }
+
+            return await customDomainRepo.save(customDomain)
+        }
+        catch (e) {
+            logger.error(e)
+            throw new ActivepiecesError({
+                code: ErrorCode.CUSTOM_DOMAIN_FAILED,
+                params: {
+                    message: 'Failed to create a custom domain. Make sure you are not using domains like *.example.com and *.example.net.',
```

Comment:
Let's remove the try -> catch, If the error is unexpected, it will return 500 and we can monitor 5xx 

Now If you are afraid of the user enters invalid custom domain, you can use the request model to add validation using sinclair that way we will fail fast even before sending to cloudflare.